msal/src/main/java/com/microsoft/identity/client/Account.java

```diff
@@ -92,6 +94,24 @@ public String getId() {
             id = mHomeOid;
         }
 
+        if (StringUtil.isEmpty(id)) {
+            com.microsoft.identity.common.logging.Logger.warn(
+                    TAG + methodName,
+                    "Unable to get account id from either ClientInfo or IdToken. Attempting to obtain from home account id."
+            );
+            id = com.microsoft.identity.common.java.util.StringUtil.getUIdFromHomeAccountId(mHomeAccountId);
+        }
+
+        if (StringUtil.isEmpty(id)) {
+            com.microsoft.identity.common.logging.Logger.warn(
+                    TAG + methodName,
+                    "Account ID is empty. Returning MISSING_FROM_THE_TOKEN_RESPONSE."
+            );
+
+            // must return something because the method is marked as non-null
```

Comment:
I think we should reconsider the fact that this method id non-null.  We should either accept the null return as a possibility, or deal with an exception, perhaps.